libgcc/config/riscv/save-restore.S

```diff
@@ -60,55 +60,107 @@ __riscv_save_12:
   # __riscv_save_* routine use t0/x5 as return address
   .cfi_return_column 5
   addi sp, sp, -112
+  .cfi_def_cfa_offset 112
   li t1, 0
   sd s11, 8(sp)
+  .cfi_offset 27, -104
   j .Ls10
 
 __riscv_save_11:
 __riscv_save_10:
+  .cfi_restore 27
   addi sp, sp, -112
+  .cfi_def_cfa_offset 112
   li t1, -16
 .Ls10:
   sd s10, 16(sp)
+  .cfi_offset 26, -96
   sd s9, 24(sp)
+  .cfi_offset 25, -88
   j .Ls8
 
 __riscv_save_9:
 __riscv_save_8:
+  .cfi_restore 25
+  .cfi_restore 26
+  .cfi_restore 27
   addi sp, sp, -112
+  .cfi_def_cfa_offset 112
   li t1, -32
 .Ls8:
   sd s8, 32(sp)
+  .cfi_offset 24, -80
   sd s7, 40(sp)
+  .cfi_offset 23, -72
   j .Ls6
 
 __riscv_save_7:
 __riscv_save_6:
+  .cfi_restore 23
+  .cfi_restore 24
+  .cfi_restore 25
+  .cfi_restore 26
+  .cfi_restore 27
   addi sp, sp, -112
+  .cfi_def_cfa_offset 112
   li t1, -48
 .Ls6:
   sd s6, 48(sp)
+  .cfi_offset 22, -64
   sd s5, 56(sp)
+  .cfi_offset 21, -56
   j .Ls4
 
 __riscv_save_5:
 __riscv_save_4:
+  .cfi_restore 21
+  .cfi_restore 22
+  .cfi_restore 24
+  .cfi_restore 25
+  .cfi_restore 26
+  .cfi_restore 27
+  .cfi_restore 24
+  .cfi_restore 25
+  .cfi_restore 26
+  .cfi_restore 27
   addi sp, sp, -112
+  .cfi_def_cfa_offset 112
   li t1, -64
 .Ls4:
   sd s4, 64(sp)
+  .cfi_offset 20, -48
   sd s3, 72(sp)
+  .cfi_offset 19, -40
   j .Ls2
 
 __riscv_save_3:
 __riscv_save_2:
+  .cfi_restore 19
+  .cfi_restore 20
+  .cfi_restore 21
+  .cfi_restore 22
+  .cfi_restore 24
+  .cfi_restore 25
+  .cfi_restore 26
+  .cfi_restore 27
+  .cfi_restore 24
+  .cfi_restore 25
+  .cfi_restore 26
+  .cfi_restore 27
   addi sp, sp, -112
+  .cfi_def_cfa_offset 112
   li t1, -80
 .Ls2:
   sd s2, 80(sp)
+  .cfi_offset 18, -32
   sd s1, 88(sp)
+  .cfi_offset 9, -24
   sd s0, 96(sp)
+  .cfi_offset 8, -16
   sd ra, 104(sp)
+  .cfi_offset 1, -8
+  # CFA info is not correct in next 2 instruction since t1's
+  # value is depend on how may register really save.
```

Comment:
If I'm understanding right this will confuse statistical profilers so it'd be somewhat useful to have working unwind here.  (I do not regard this as upstream-blocking to be clear)  Would fixing this require an arch-specific unwind opcode?